elements/geosearch/main.js

```diff
@@ -0,0 +1,375 @@
+import { LitElement, html, css } from "lit";
+import Handlebars from "handlebars";
+import proj4 from "proj4";
+import _debounce from "lodash.debounce";
+
+import { button } from "../../utils/styles/button";
+
+import styles from "./style";
+
+class EOxGeoSearch extends LitElement {
+  static get properties() {
+    return {
+      /**
+       * Internal storage of OpenCage API data after a successful API request.
+       *
+       */
+      _data: { attribute: false },
+      /**
+       * Whether or not the list dropdown is visible.
+       *
+       */
+      _isListVisible: { attribute: false },
+      /**
+       * Whether or not the input field is visible.
+       *
+       */
+      _isInputVisible: { attribute: false },
+      /**
+       * The search query, which is bound to the input field.
+       *
+       */
+      _query: { attribute: false },
+      _useMockData: { attribute: false },
+      /**
+       * The OpenCage API endpoint to use for the search, as a Handlebars template string with `query` and `key` variables.
+       *
+       */
+      endpoint: { type: String },
+      /**
+       * The API key to use for the search.
+       *
+       */
+      key: { type: String },
+      /**
+       * The query selector for the map target, if any.
+       *
+       */
+      for: { type: String },
+      /**
+       * A function to be called when a search result is selected, to enable
+       * the parent component to handle the selection.
+       *
+       */
+      onSelect: { type: Function },
+      /**
+       * Whether or not to enable button mode, which hides and shows the input field
+       * similar to how a modal works.
+       *
+       */
+      button: { type: Boolean },
+      /**
+       * Limit the search results to a certain number of items.
+       *
+       */
+      limit: { type: Number, default: 10 },
+      /**
+       * Set a custom interval for the debounce function.
+       *
+       */
+      interval: { type: Number, default: 200 },
+      /**
+       * Enables a smaller version of the button for use in map controls.
+       *
+       */
+      small: { type: Boolean },
+      /**
+       * Which text to use for the button if it is enabled.
+       *
+       */
+      label: { type: String },
+      /**
+       * The direction of the search input relative to the button, with the following options:
+       *
+       * - `left`
+       * - `top`
+       * - `right`
+       * - `bottom`
+       *
+       */
+      direction: { type: String },
+      /**
+       * The direction of the results box relative to the input, with the following options:
+       *
+       * - `left`
+       * - `top`
+       * - `right`
+       * - `bottom`
+       *
+       */
+      resultsDirection: { type: String, default: "rightx" },
```

Comment:
In general, attributes can't have camelCase - needs e.g. `attribute: "results-direction"`